docs/docs/guides/labs/components/using-environment-variables-in-components.md

```diff
@@ -0,0 +1,89 @@
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/1-dg-init.txt" />
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/2-activate-venv.txt" />
+
+We'll install `dagster-sling` and scaffold an empty Sling connection component:
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/3-uv-add-sling.txt" />
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/4-dg-list-component-types.txt" />
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/5-dg-scaffold-sling.txt" />
+
+## Utilizing environment variables in a component
+
+Next, we will configure a Sling connection which will sync a local CSV file to a Snowflake database, whose credentials will be provided via environment variables.
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/6-curl.txt" />
+
+<CodeExample path="docs_snippets/docs_snippets/guides/dg/using-env/7-replication.yaml" language="yaml" title="src/ingestion/defs/ingest_files/replication.yaml" />
+
+We will use the `env` function to template credentials into Sling configuration in our `component.yaml` file. Running `dg check yaml` will highlight that we
+need to explicitly encode these environment dependencies at the bottom of the file:
+<CodeExample path="docs_snippets/docs_snippets/guides/dg/using-env/8-component.yaml" language="yaml" title="src/ingestion/defs/ingest_files/component.yaml" />
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/9-dg-component-check.txt" />
+
+After adding the environment dependencies, running `dg check yaml` again will confirm that the file is valid:
+<CodeExample path="docs_snippets/docs_snippets/guides/dg/using-env/10-component-with-env-deps.yaml" language="yaml" title="src/ingestion/defs/ingest_files/component.yaml" />
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/11-dg-component-check-fixed.txt" />
+
+Next, you can invoke `dg list env`, which shows all environment variables configured or used by components in the project. Here we can see all of the Snowflake credentials we must configure in our shell or `.env` file in order to run our project:
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/12-dg-list-env.txt" />
+
+You can edit the `.env` file in your project root to specify environment variables for Dagster to use when running the project locally. You can run `dg list env` again to see that they are now set:
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/13-inject-env.txt" />
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/using-env/14-dg-list-env.txt" />
+
+## Configuring environment variables for components in Dagster Plus
+
+If you are using Dagster Plus, you can also use the `dg` CLI to push environment variables to your deployment.
```

Comment:
```suggestion
If you are using Dagster+,  you can also use the `dg` CLI to push environment variables to your deployment.
```

Reply:
I would add a line about why this is helpful here -- I assume so you can have different creds for production vs local environments.